src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqTransportBindingElement.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Net;
+using System.ServiceModel.Channels;
+using RabbitMQ.Client;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqTransportBindingElement : TransportBindingElement
+    {
+        private const int MaxRabbitMqMessageSize = 536870912; // 512 * (2^20) = 512 MB, max message size as of RabbitMQ v3.8
+        private const int DefaultMaxMessageSize = 65535;  // 64K
+        private long _maxReceivedMessageSize = DefaultMaxMessageSize;
+
+        public RabbitMqTransportBindingElement()
+        { }
+
+        private RabbitMqTransportBindingElement(RabbitMqTransportBindingElement other)
+            : base(other)
+        {
+            MaxReceivedMessageSize = other.MaxReceivedMessageSize;
+            BaseAddress = other.BaseAddress;
+            BrokerProtocol = other.BrokerProtocol;
+            SslOption = other.SslOption;
+            VirtualHost = other.VirtualHost;
+            Credentials = other.Credentials;
+        }
+
+        public Uri BaseAddress { get; internal set; }
+
+        /// <summary>
+        /// Gets the scheme used by the binding, soap.amqp
+        /// </summary>
+        public override string Scheme
+        {
+            get { return RabbitMqDefaults.Scheme; }
+        }
+
+        /// <summary>
+        /// The largest receivable encoded message
+        /// </summary>
+        public override long MaxReceivedMessageSize
+        {
```

Comment:
You don't need this property here. The property exists on the service side so you can limit resource allocation to prevent malicious clients. You don't need to worry about that on the client side. You can still have a check when serializing as you know it will fail if it's bigger than 512MB as RabbitMQ can't accept messages bigger than that, but you don't need to allow it to be configurable on the client side. If you want to allow it to be configurable, then you should drop Received from the name as you aren't receiving anything.